### ISSUE TITLE
subclassing histogram filter crash 

### ISSUE THREAD

**Issue by kallipigous** [BradLarson/GPUImage#2304]
If I subclass the histogram filter it works fine until I overwrite
- (void)renderToTextureWithVertices:(const GLfloat *)vertices textureCoordinates

even if I duplicate the original it crashes in
- (GPUImageFramebuffer *)fetchFramebufferForSize:(CGSize)framebufferSize textureOptions:(GPUTextureOptions)textureOptions onlyTexture:(BOOL)onlyTexture;

is there anything I can do to prevent that?

**Comment by BradLarson** [comment]
When you say it crashes, where does it crash and what do you see on the console? I'm betting you're trying to create a framebuffer at an illegal size (height lower than 3 pixels, for example).

**Comment by kallipigous** [comment]
Oh Hello Brad

No I’m creating something at 256x256, This is a vectorscope incidentally.

it looks to me that as soon as I overwrite renderToTerxture it doesn’t know what self.outpyuttextureoptions is.

I’m just trying to remove the GL_FUNC_ADD

remove renderToTexture and it works fine

Add it back and it crashes

**Comment by kallipigous** [comment]
If I subclass GPUImageFilter and cut and paste all the methods from the histogram (avoiding the duplicate Shader Strings)  it crashes in a slightly different place.

```
[framebufferFromCache lock];
```

[__NSCFData lock]: unrecognized selector sent to instance 0x15f824ef0

**Comment by kallipigous** [comment]
I have found a work around by editing the original histogramFilter and moving the drawing elements to a new method and overwriting that. It’s just the framebuffer part that seems to cause issues. Is it a thread issue?

Every time I update cocopods it’s going to break again but this is great for the time being and gives me full control of the blending.

**Comment by ryanstout** [comment]
@kallipigous did you ever manage to get your vectorscope working?  I'm seeing crashes with the histogram also.

**Comment by kallipigous** [comment]
I did when I edited the histogram and moved the drawing parts into a separate function.

I never figured out why but editing it worked.

**Comment by ryanstout** [comment]
@kallipigous thanks for getting back to me.  So you actually changed the class in the library?

**Comment by ryanstout** [comment]
@kallipigous what did you use the vectorscope for?  (Just curious)

**Comment by kallipigous** [comment]
Exactly. Then I found I could subclass it fine. The histogram worked off the bat but not subclasses.

**Comment by kallipigous** [comment]
An app for cinematographers called Artemis

**Comment by omxcodec** [comment]
@kallipigous have you turned off the  ARC？NO ARC will cause crash.

((void)renderToTextureWithVertices:(const GLfloat *)vertices textureCoordinates) 
my git version: 167b0389bc6e9dc4bb0121550f91d8d5d6412c53

maybe you should add "-fobjc-arc" flags for you filter souce file.
